ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4298,7 +4298,7 @@ private List<Partition> getPartitionsWithAuth(Table tbl, Map<String, String> par
   public List<Partition> getPartitions(Table tbl, Map<String, String> partialPartSpec)
   throws HiveException {
     if (tbl.getStorageHandler() != null && tbl.getStorageHandler().alwaysUnpartitioned()) {
-      return tbl.getStorageHandler().getPartitions(tbl, partialPartSpec);
+      return tbl.getStorageHandler().getPartitions(tbl, partialPartSpec, false);
```

Comment:
same question, should it be false? you changed the other one

Reply:
No, because it is called from compaction flow when compaction request is called with a specific partition which can be partition of an old partition spec: https://github.com/apache/hive/blob/76a03b8173e36681d8824bd30920cadff0bfb4fa/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java#L143-L147